qp_klp/klp.py

```diff
@@ -32,32 +44,228 @@ def sequence_processing_pipeline(qclient, job_id, parameters, out_dir):
     bool, list, str
         The results of the job
     """
-    qclient.update_job_step(job_id, "Step 1 of 3: Collecting information")
     run_identifier = parameters.pop('run_identifier')
     sample_sheet = parameters.pop('sample_sheet')
+    job_pool_size = 30
+
+    # checking if this is running as part of the unittest
+    # https://stackoverflow.com/a/25025987
+    skip_exec = True if [x for x in stack() if
+                         'unittest' in x.filename] else False
 
     success = True
     ainfo = None
     msg = None
-    if exists(run_identifier) and isdir(run_identifier):
```

Comment:
is this really no longer required? why?

Reply:
It was appropriate before when run_identifier was being used to hold the full path to a run-dir, but the correct operation is to supply a run_id, and the Pipeline() object will determine the path based on its configured search paths. In this case, Pipeline() is already performing those two checks internally.